pkg/shared/backup.go

```diff
@@ -103,31 +103,40 @@ func (b *BackupShared) BackupRun(backupRoutine *model.BackupRoutine, backupPolic
 	setCString(&backupConfig.s3_region, storage.S3Region)
 	setCString(&backupConfig.s3_profile, storage.S3Profile)
 
+	result := &BackupStat{}
 	if isIncremental {
-		// for incremental backup
 		setCLong(&backupConfig.mod_after, opts.ModAfter)
-		setCString(&backupConfig.output_file, getIncrementalPath(storage))
+		path := getIncrementalPath(storage)
+		result.Path = path
+		setCString(&backupConfig.output_file, path)
 	} else {
-		// for full backup
-		setCString(&backupConfig.directory, getPath(storage, backupPolicy))
+		path := getPath(storage, backupPolicy)
+		result.Path = path
+		setCString(&backupConfig.directory, path)
 	}
 
 	backupStatus := C.backup_run(&backupConfig)
+	// destroy the backup_config
+	defer C.backup_config_destroy(&backupConfig)
 
-	var success bool
-	if unsafe.Pointer(backupStatus) == C.RUN_BACKUP_SUCCESS { //nolint:gocritic
-		success = true
-	} else if unsafe.Pointer(backupStatus) != C.RUN_BACKUP_FAILURE {
-		C.backup_status_destroy(backupStatus)
-		C.cf_free(unsafe.Pointer(backupStatus))
-		success = true
-	} else {
+	if unsafe.Pointer(backupStatus) == C.RUN_BACKUP_FAILURE {
 		slog.Warn("Failed backup operation", "policy", backupRoutine.Name)
+		return nil
 	}
 
-	// destroy the backup_config
-	C.backup_config_destroy(&backupConfig)
-	return success
+	if unsafe.Pointer(backupStatus) == C.RUN_BACKUP_SUCCESS { //nolint:gocritic
+		return result
+	}
+
+	result.HasStats = true
+	result.RecordCount = int(backupStatus.rec_count_total)
+	result.SecondaryIndexCount = int(backupStatus.index_count)
+	result.UDFFileCount = int(backupStatus.udf_count)
```

Comment:
nit: can be taken out to a function

Reply:
I'm afraid I cannot, I don't know the type of backupStatus